packages/models-library/tests/test_basic_regex.py

```diff
@@ -100,3 +102,31 @@ def test_pep404_compare_versions():
     assert Version("1.9.a.dev") == Version("1.9a0dev")
     assert Version("2.1-rc2") < Version("2.1")
     assert Version("0.6a9dev") < Version("0.6a9")
+
+
+@pytest.mark.parametrize(
+    "string_under_test",
+    (
+        "x1",
+        "a_very_long_variable_22",
+        "a-string-with-minus",
+        "1xxx",
+        "str w/ spaces",
+        "a./b.c",
+        "def",
+        "for",
+        "in",
+    ),
+)
+def test_variable_names_regex(string_under_test):
```

Comment:
`_start-with-underscore` should also fail because it cannot be a variable name

Reply:
with respect to starting with ``_`` ... i do not think is a good idea to allow protected/private variables. I will change the nae of the regex to ``PUBLIC_VARIABLE_NAME_RE``

The idea here is to have a regex that can validate a name in e.g. a port name, so we can guarantee we can use it as an argument variable in the underling function (this will be used for function nodes)

We do not want to have argument variables called ``_``, right? Because this is in addition reserved !

Reply:
`_` is just a convention, it is not reserved.
You can either:
- make sure it is supported
- or remove support for it